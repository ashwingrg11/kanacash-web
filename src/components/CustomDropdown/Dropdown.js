import React, { useState, useEffect, useRef } from "react";
import dropdownArrow from "./dropdownArrow.svg";
import {
  MainContainer,
  DropDownContainer,
  DropDownWithoutTextInputHeader,
  DropDownListContainer,
  Row,
  DropDownList,
  ListItem,
  Label,
  Button,
} from "./StyledComponent";
import PropTypes from "prop-types";
import { isEmpty } from "../../utils/isEmpty";

Dropdown.defaultProps = {
  containerWidth: "20rem",
};

export default function Dropdown({
  /**
   * list of array: dropdown list
   */
  option,
  /**
   * text lebel for the dropdown option
   */
  label,
  /**
   * width for the container
   */
  onClick,
  /**
   * function to get selectedPicker object in parent
   */
  onChangeOption,
  pickerOption,
  containerWidth,
  customName,
  customValue,

  placeHolder,
}) {
  const wrapperRef = useRef(null);
  useOutsideAlerter(wrapperRef);
  const [isOpen, setIsOpen] = useState(false);
  // const [selectedOption, setSelectedOption] = useState(null);

  const toggling = () => setIsOpen(!isOpen);

  const onOptionClicked = (value) => () => {
    // setSelectedOption(value);
    setIsOpen(false);
    let returnValue = {
      name: customName,
      value: value,
    };
    onChangeOption(returnValue);
  };

  /**
   * Hook that hide dropdown list if clicks outside of the passed ref
   */
  function useOutsideAlerter(ref) {
    useEffect(() => {
      /**
       * hide if clicked on outside of element
       */
      function handleClickOutside(event) {
        if (ref.current && !ref.current.contains(event.target)) {
          setIsOpen(false);
        }
      }

      // Bind the event listener
      document.addEventListener("mousedown", handleClickOutside);
      return () => {
        // Unbind the event listener on clean up
        document.removeEventListener("mousedown", handleClickOutside);
      };
    }, [ref]);
  }

  return (
    <MainContainer ref={wrapperRef}>
      <DropDownContainer className="input-dropdown" style={{ width: "100%" }}>
        <DropDownWithoutTextInputHeader
          onClick={toggling}
          className="custom-wrap"
        >
          {label && <Label>{label}</Label>}
          <Row>
            <div className="selected-country">
              {customValue?.flagUrl && (
                <span>
                  <img
                    alt=""
                    src={customValue?.flagUrl}
                    style={{ marginRight: "0px" }}
                  />
                </span>
              )}
              {!isEmpty(customValue) ? (
                customValue[pickerOption]
              ) : (
                <span>{placeHolder}</span>
              )}
            </div>
            <img alt="arrow-dwon" src={dropdownArrow} />
          </Row>
        </DropDownWithoutTextInputHeader>
        {isOpen && (
          <DropDownListContainer className="show-country-dd">
            <DropDownList>
              {option.map((item) => (
                <ListItem onClick={onOptionClicked(item)} key={Math.random()}>
                  {item?.flagUrl && (
                    <span>
                      <img
                        alt=""
                        src={item.flagUrl}
                        height="28px"
                        width="28px"
                        style={{ marginRight: 10 }}
                      />
                    </span>
                  )}
                  {item[pickerOption]}
                </ListItem>
              ))}
            </DropDownList>
          </DropDownListContainer>
        )}
      </DropDownContainer>
      {onClick && <Button onClick={onClick}>Send</Button>}
    </MainContainer>
  );
}

Dropdown.propTypes = {
  option: PropTypes.array.isRequired,
  label: PropTypes.string,
  onClick: PropTypes.func,
  onChangeOption: PropTypes.func,
  pickerOption: PropTypes.string,
  containerWidth: PropTypes.string,
  customValue: PropTypes.object,
  customName: PropTypes.string,
  placeHolder: PropTypes.string,
};
