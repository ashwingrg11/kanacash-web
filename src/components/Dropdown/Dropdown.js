import React, { Component } from "react";
import { Form } from "react-bootstrap";
import PropTypes from "prop-types";
let name;

class Dropdowns extends Component {
  constructor(props) {
    super(props);
    this.state = {
      padding: "5px",
    };
  }

  componentDidMount = () => {
    //   name=item.this.props.itemsName
    if (this.props.isIcon) {
      this.setState({ padding: "50px" });
    }
  };

  render() {
    return (
      // this.props.options !== undefined && (
      <Form.Control
        as="select"
        id={this.props.id}
        onChange={this.props.handleChange}
        style={{ height: "50px", paddingLeft: this.state.padding }}
        value={this.props.value?.value}
      >
        <option value="" style={{ height: "30px" }}>
          {this.props.title}
          {name}
        </option>
        {this.props.options !== undefined &&
          this.props.options.map((item, index) => {
            return (
              <option value={item[this.props.pickerValue]} key={index}>
                {`${item[this.props.optionValue]} `}{" "}
                {this.props.optionValue2 && item[this.props?.optionValue2]}
              </option>
            );
          })}
        {this.props.loading && (
          <option value="" style={{ height: "30px" }} disabled>
            LOADING...
          </option>
        )}
      </Form.Control>
      // )
    );
  }
}

Dropdowns.defaultProps = {
  pickerValue: "id",
};

Dropdowns.propTypes = {
  isIcon: PropTypes.bool,
  id: PropTypes.string,
  handleChange: PropTypes.func,
  value: PropTypes.object,
  title: PropTypes.string,
  options: PropTypes.array,
  optionValue: PropTypes.string,
  pickerValue: PropTypes.string.isRequired,
  optionValue2: PropTypes.string,
  loading: PropTypes.bool,
};
export default Dropdowns;
