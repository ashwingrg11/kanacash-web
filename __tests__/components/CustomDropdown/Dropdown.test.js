import React from "react";
import Dropdown from "../../../src/components/CustomDropdown/Dropdown";
import { shallow } from "enzyme";
// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";

describe("<Dropdown />", () => {
  it("Dropdown renders correctly", () => {
    const option =[]
    const tree = renderer.create(<Dropdown option={option}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
