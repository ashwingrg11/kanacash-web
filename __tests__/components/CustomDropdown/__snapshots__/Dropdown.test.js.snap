// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> Dropdown renders correctly 1`] = `
<div
  className="sc-AxjAm jhiTqR"
>
  <div
    className="sc-AxirZ fpkyRk input-dropdown"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="sc-AxhCb flAYhK custom-wrap"
      onClick={[Function]}
    >
      <div
        className="sc-AxgMl fqhLTf"
      >
        <div
          className="selected-country"
        >
          <span />
        </div>
        <img
          alt="arrow-dwon"
          src=""
        />
      </div>
    </div>
  </div>
</div>
`;
